### PR TITLE
fix(repository): remove property.array() call from hasMany decorator

### LINKED ISSUE #1944: Create related model in hasMany relation inconsistency
### Description / Steps to reproduce / Feature proposal
I have 2 Models: `Profile` and `Team`. The relation is `Team hasMany Profile`.

I have a controller for that relation **TeamProfilesController**
```ts
export class TeamProfilesController {
  constructor(
    @repository(TeamRepository) protected teamRepository: TeamRepository,
  ) {}

  @post('/teams/{id}/members')
  async createMember(
    @param.path.string('id') teamId: typeof Team.prototype.id,
    @requestBody() profileData: Profile,
  ): Promise<Profile> {
    return await this.teamRepository.members(teamId).create(profileData);
  }

  @get('/teams/{id}/members')
  async getMember(
    @param.path.string('id') teamId: typeof Team.prototype.id,
    @param.query.object('filter', getFilterSchemaFor(Profile)) filter?: Filter,
  ): Promise<Profile[]> {
    return await this.teamRepository.members(teamId).find(filter);
  }
}
```

If I already have a team created and use POST /teams/{id}/members, a new profile is created :+1: 

### Current Behavior

The problem is that on my team repository the POST /teams receives:
```json
{
  "id": "string",
  "name": "string",
  "type": "string",
  "members": [
    {
      "id": "string",
      "username": "string",
      "password": "string",
      "name": "string",
      "email": "string"
    }
  ]
}
```

The repository accepts `members` as valid property and if I create a team with the property `members` set, what ends up in my Database is:
```json
{
    "_id" : ObjectId("5bd83d9ae8abbe501d4b68d0"),
    "name" : "testTeam",
    "members" : [ 
        {
            "id" : "should run function Profilerepository.create that creates a new ObjectId",
            "username" : "test user",
            "password" : "test pass",
            "name" : "test name",
            "email" : "testemail@email.com"
        }
    ]
}
```
No new Profile is created. What is created is a Team that has property members embedded in it. And the objects on that array don't qualify as Profile.
When I try to GET /teams/5bd83d9ae8abbe501d4b68d0/members is returns empty array.

### Expected Behavior
When I create a team with the property `members` I Expected 1 of 2.
**1**. One Team and one profile is created.
```json
{
    "_id" : ObjectId("5bd83d9ae8abbe501d4b68d0"),
    "name" : "testTeam"
}
{
  "id" : "should run function Profilerepository.create that creates a new ObjectId",
  "username" : "test user",
  "password" : "test pass",
  "name" : "test name",
  "email" : "testemail@email.com"
}
```
**2**. Like LB3 to not be possible to create a Team with the members property on the POST /teams. And if the property is sent, so get caught by the model as an invalid property because it is a "strict" model.

### PR DESCRIPTION
**Pull request by SharonItz**
…y relation inconsistency

Fix for issue #1944 - Create related model in hasMany relation inconsistency

fix #1944

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/repository/test/acceptance/has-many.relation.acceptance.ts [hunk 1]
```diff
@@ -139,7 +139,13 @@ describe('HasMany relation', () => {
     expect(_.pick(orders[0], ['customerId', 'description'])).to.eql(newOrder);
   });
 
-  // This should be enforced by the database to avoid race conditions
+    it("does not create an array of the related model"), async() => {
```
Comment (b-admike) [inline packages/repository/test/acceptance/has-many.relation.acceptance.ts]: We should remove the extra `)` after the title of the test here.

**Comment by b-admike** [comment]
In terms of the commit message, I highly suggest a descriptive message for what it is you are doing in this PR (like `remove property.array() call from hasMany decorator`). The `fix(repository)` part already signifies that it is a fix for a bug, so we can leave out `Fix for issue #1944` from your message header/title (feel free to point it out in the commit message body though). 
```
Fix for issue #1944 - Create related model in hasMany relation inconsistency

fix #1944 
```

**Comment by SharonItz** [comment]
> In terms of the commit message, I highly suggest a descriptive message for what it is you are doing in this PR (like `remove property.array() call from hasMany decorator`). The `fix(repository)` part already signifies that it is a fix for a bug, so we can leave out `Fix for issue #1944` from your message header/title (feel free to point it out in the commit message body though).


Hi @b-admike 
Thanks for your help and the comment,
BTW, This is my first PR 😊 and the first time that I’m using the “commitizen” tool. I will apply this commit guide on the next PR.
Thanks, Sharon

File: packages/repository-json-schema/test/integration/build-schema.integration.ts [hunk 2]
```diff
@@ -457,22 +457,9 @@ describe('build-schema', () => {
           });
           expect(customerSchema.properties).to.deepEqual({
             id: {type: 'number'},
-            orders: {
-              type: 'array',
-              items: {$ref: '#/definitions/Order'},
-            },
-          });
-          expect(customerSchema.definitions).to.deepEqual({
```
Comment (bajtos) [inline packages/repository-json-schema/test/integration/build-schema.integration.ts]: I feel we should rework this check instead of removing it.

How about the following assertion?

```ts
expect(Object.keys(customerSchema.definitions)).to.not.containEql('Order');
```

File: packages/repository/src/relations/has-many/has-many.decorator.ts [hunk 3]
```diff
@@ -34,6 +32,9 @@ export function hasMany<T extends Entity>(
         source: decoratedTarget.constructor,
         target: targetResolver,
       },
+
+
+
```
Comment (bajtos) [inline packages/repository/src/relations/has-many/has-many.decorator.ts]: Extra white space, please remove. I am surprised the linter did no complain.

File: packages/repository/test/unit/decorator/relation.decorator.unit.ts [hunk 4]
```diff
@@ -54,11 +54,6 @@ describe('relation decorator', () => {
         target: () => Address,
       });
 
-      expect(jugglerMeta).to.eql({
```
Comment (bajtos) [inline packages/repository/test/unit/decorator/relation.decorator.unit.ts]: Similarly here - let's rework this check to assert that no property was defined by the relation decorator.

File: packages/repository/test/unit/decorator/relation.decorator.unit.ts [hunk 5]
```diff
@@ -100,33 +95,9 @@ describe('relation decorator', () => {
         target: () => Address,
         keyTo: 'someForeignKey',
       });
-      expect(jugglerMeta).to.eql({
```
Comment (bajtos) [inline packages/repository/test/unit/decorator/relation.decorator.unit.ts]: ditto

File: packages/repository-json-schema/test/integration/build-schema.integration.ts (hunk 2 above)
Comment (SharonItz) [reply packages/repository-json-schema/test/integration/build-schema.integration.ts]: Hi @bajtos thanks for your comments.
The original tests were added and changed accordantly.
Code was reformatted using lint (npm run lint:fix).

File: packages/repository/src/relations/has-many/has-many.decorator.ts (hunk 3 above)
Comment (SharonItz) [reply packages/repository/src/relations/has-many/has-many.decorator.ts]: Code was reformatted using lint (npm run lint:fix).

File: packages/repository/test/unit/decorator/relation.decorator.unit.ts (hunk 4 above)
Comment (SharonItz) [reply packages/repository/test/unit/decorator/relation.decorator.unit.ts]: The original tests were added and changed accordantly

File: packages/repository/test/unit/decorator/relation.decorator.unit.ts (hunk 5 above)
Comment (SharonItz) [reply packages/repository/test/unit/decorator/relation.decorator.unit.ts]: The original tests were added and changed accordantly

**Comment by b-admike** [comment]
@SharonItz Thanks for applying all the feedback. I've squashed and rebased the commits for you so we can land it now. @bajtos if this LGTY, feel free to merge 👍.

File: packages/repository-json-schema/test/integration/build-schema.integration.ts [hunk 6]
```diff
@@ -527,57 +532,52 @@ describe('build-schema', () => {
...
+  }
+});
 
-    it('creates JSON schema if one does not already exist', () => {
-      @model()
-      class NewModel {
-        @property()
-        newProperty: string;
-      }
+describe('getjsonSchema', () => {
```
Comment (bajtos) [inline packages/repository-json-schema/test/integration/build-schema.integration.ts]: IIUC, this change is moving `getjsonSchema` test suite from `build-schema` suite to top-level. I'd rather avoid such unrelated change.

**Comment by SharonItz** [comment]
Hi  @b-admike, @bajtos and the team,
Thank you for the PR approval. :)
